cmd/dep/init.go

```diff
@@ -152,7 +163,20 @@ func (cmd *initCommand) Run(ctx *dep.Ctx, args []string) error {
 	l = dep.LockFromSolution(soln)
 
 	rootAnalyzer.FinalizeRootManifestAndLock(m, l)
-	gs.FinalizeRootManifestAndLock(m, l)
+	if cmd.gopath {
+		gs.FinalizeRootManifestAndLock(m, l)
+	} else {
+		// Pick the direct dependencies from the above solution and add to manifest
```

Comment:
>The logic should be in rootAnalyzer.FinalizeManifestAndLock instead of the init command directly.

Yeah, I felt I made init dirty. It became elegant with all the new stuff and with all these changes, it felt dirty 😅 I'll change that.

> When someone runs dep init -skip-tools -gopath it shouldn't result in a populated lock and an empty manifest.

We have Integration/init/case3 to handle this scenarios 😊